### PR TITLE
ci: Revamp workflows to run with the configurations in mind and remove unnecessary support GitHub Actions.

### PR DESCRIPTION
**Pull request by KirilMihaylov**

### CODE REVIEW

File: .github/test-data/build-configuration/protocol.json
Comment (gmanev7) [inline .github/test-data/build-configuration/protocol.json:1]: do we still need protocol and topology definitions for the tests?

Comment (KirilMihaylov) [reply .github/test-data/build-configuration/protocol.json:1]: Yes, we do in the case when `testing` is not turned on, which is a non-negligible portion from what I remember. I don't expect us changing it as long as we don't change the topology, which too I hope is unlikely.

Comment (KirilMihaylov) [reply .github/test-data/build-configuration/protocol.json:1]: Not to mention that this also does indirectly test the currency generation too.

Comment (gmanev7) [reply .github/test-data/build-configuration/protocol.json:1]: expressed offline my concerns over usage of these dummy currencies in packages that depend on "currencies", like leaser, oracle, lpp, to mention a few

Comment (gmanev7) [reply .github/test-data/build-configuration/protocol.json:1]: validated that they are copied in lint, test and check stages

**Pull request merged**